server/amp/handlers/infinite-scroll.js

```diff
@@ -20,11 +20,19 @@ async function storyPageInfiniteScrollHandler(req, res, next, { client, config,
     infiniteScrollSource,
   });
   const jsonResponse = await infiniteScrollAmp.getResponse();
-  if (jsonResponse instanceof Error) return next(jsonResponse);
+  if (jsonResponse instanceof Error) return handleErrorResponse(res, next, jsonResponse);
   res.set("Content-Type", "application/json; charset=utf-8");
   setCorsHeaders({ req, res, next, publisherConfig: config });
 
   if (!res.headersSent) return res.send(jsonResponse);
 }
 
+function handleErrorResponse(res, next, error) {
+  const errorMessage = error.message;
+  if (errorMessage === `Query param "story-id" missing`) {
+    return res.status(400).send(errorMessage);
+  }
+  return next(errorMessage);
+}
+
```

Comment:
instead of this, just check if `req.query["story-id"]` is present here itself

Reply:
done